diff --git a/lib/src/imdb/db_binary.rs b/lib/src/imdb/db_binary.rs
index b3428fe..261af65 100644
--- a/lib/src/imdb/db_binary.rs
+++ b/lib/src/imdb/db_binary.rs
@@ -1,8 +1,11 @@
 #![warn(clippy::all)]
 
+use std::path::{Path, PathBuf};
+
 use crate::imdb::db::{Db, Query};
 use crate::imdb::title::Title;
 use crate::imdb::title_id::TitleId;
+use crate::utils::io::file as io_file;
 use crate::utils::search::SearchString;
 
 use log::debug;
@@ -16,13 +19,38 @@ pub enum Error {
   /// Loading title error.
   #[error("Error loading title: {0}")]
   LoadingTitle(#[from] crate::imdb::title::Error),
+  /// File-related error.
+  #[error("File handling error: {0}")]
+  File(#[from] crate::utils::io::file::Error),
 }
 
-pub struct ServiceDbFromBinary {
+pub(crate) struct DbBinaryBuilder {
+  pub(crate) movies_db_filename: PathBuf,
+  pub(crate) series_db_filename: PathBuf,
+}
+
+impl DbBinaryBuilder {
+  pub(crate) fn new(cache_dir: &Path) -> Self {
+    let movies_db_filename = cache_dir.join(DbBinary::MOVIES_DB_FILENAME);
+    let series_db_filename = cache_dir.join(DbBinary::SERIES_DB_FILENAME);
+    Self { movies_db_filename, series_db_filename }
+  }
+
+  pub(crate) fn build(self) -> Result<DbBinary, Error> {
+    let movies_data = io_file::read_static(&self.movies_db_filename)?;
+    let series_data = io_file::read_static(&self.series_db_filename)?;
+    DbBinary::new(movies_data, series_data)
+  }
+}
+
+pub struct DbBinary {
   dbs: Vec<Db>,
 }
 
-impl ServiceDbFromBinary {
+impl DbBinary {
+  const MOVIES_DB_FILENAME: &str = "imdb-movies.tvrankdb";
+  const SERIES_DB_FILENAME: &str = "imdb-series.tvrankdb";
+
   /// Load titles from the given binary data.
   ///
   /// Loads titles from the provided binary content buffers (`movies_data` and
@@ -32,7 +60,7 @@ impl ServiceDbFromBinary {
   ///
   /// * `movies_data` - Binary movies data.
   /// * `series_data` - Binary series data.
-  pub(crate) fn new(mut movies_data: &'static [u8], mut series_data: &'static [u8]) -> Result<Self, Error> {
+  fn new(mut movies_data: &'static [u8], mut series_data: &'static [u8]) -> Result<Self, Error> {
     let nthreads = rayon::current_num_threads();
     let dbs = const_mutex(Vec::with_capacity(nthreads));
     let movies_cursor: Mutex<&mut &'static [u8]> = const_mutex(&mut movies_data);
@@ -233,13 +261,13 @@ impl ServiceDbFromBinary {
 #[cfg(test)]
 mod tests {
   use crate::imdb::db::Query;
-  use crate::imdb::db_binary::ServiceDbFromBinary;
+  use crate::imdb::db_binary::DbBinary;
   use crate::imdb::testdata::{make_basics_reader, make_ratings_reader};
   use crate::imdb::title_id::TitleId;
   use crate::imdb::tsv_import::tsv_import;
   use crate::utils::search::SearchString;
 
-  fn make_service_db_from_binary() -> ServiceDbFromBinary {
+  fn make_service_db_from_binary() -> DbBinary {
     let basics_reader = make_basics_reader();
     let ratings_reader = make_ratings_reader();
 
@@ -249,7 +277,7 @@ mod tests {
 
     let movies_storage = Box::leak(movies_storage.into_boxed_slice());
     let series_storage = Box::leak(series_storage.into_boxed_slice());
-    ServiceDbFromBinary::new(movies_storage, series_storage).unwrap()
+    DbBinary::new(movies_storage, series_storage).unwrap()
   }
 
   #[test]
diff --git a/lib/src/imdb/service.rs b/lib/src/imdb/service.rs
index 7e5d920..ed9972b 100755
--- a/lib/src/imdb/service.rs
+++ b/lib/src/imdb/service.rs
@@ -4,7 +4,7 @@ use std::path::Path;
 use std::time::{Duration, Instant};
 
 use crate::imdb::db::Query;
-use crate::imdb::db_binary::ServiceDbFromBinary;
+use crate::imdb::db_binary::{DbBinary, DbBinaryBuilder};
 use crate::imdb::title::Title;
 use crate::imdb::title_id::TitleId;
 use crate::imdb::tsv_import::tsv_import;
@@ -39,16 +39,13 @@ pub enum Error {
 
 /// Struct providing the movies and series databases and the related services.
 pub struct Service {
-  service_db: ServiceDbFromBinary,
+  service_db: DbBinary,
 }
 
 const IMDB_URL: &str = "https://datasets.imdbws.com/";
 const BASICS_FILENAME: &str = "title.basics.tsv.gz";
 const RATINGS_FILENAME: &str = "title.ratings.tsv.gz";
 
-const MOVIES_DB_FILENAME: &str = "imdb-movies.tvrankdb";
-const SERIES_DB_FILENAME: &str = "imdb-series.tvrankdb";
-
 impl Service {
   /// Returns a Service struct holding movies/series databases
   ///
@@ -64,18 +61,18 @@ impl Service {
   ) -> Result<Self, Error> {
     let one_month = Duration::from_secs(60 * 60 * 24 * 30);
 
-    let movies_db_filename = cache_dir.join(MOVIES_DB_FILENAME);
-    let series_db_filename = cache_dir.join(SERIES_DB_FILENAME);
-    Self::ensure_db_files(&movies_db_filename, &series_db_filename, one_month, force_db_update, progress_fn)?;
+    let db_builder = DbBinaryBuilder::new(cache_dir);
+    Self::ensure_db_files(
+      &db_builder.movies_db_filename,
+      &db_builder.series_db_filename,
+      one_month,
+      force_db_update,
+      progress_fn,
+    )?;
 
     let start = Instant::now();
-    let movies_data = io_file::read_static(&movies_db_filename)?;
-    let series_data = io_file::read_static(&series_db_filename)?;
-    debug!("Read IMDB database in {}", format_duration(Instant::now().duration_since(start)));
-
-    let start = Instant::now();
-    let service = Self { service_db: ServiceDbFromBinary::new(movies_data, series_data)? };
-    debug!("Parsed IMDB database in {}", format_duration(Instant::now().duration_since(start)));
+    let service = Self { service_db: db_builder.build()? };
+    debug!("Loaded IMDB database in {}", format_duration(Instant::now().duration_since(start)));
 
     if log_enabled!(log::Level::Debug) {
       let (total_movies, total_series) = service.service_db.n_entries();
